libgearman-server/plugins/queue/sqlite/queue.cc

```diff
@@ -88,16 +88,11 @@ Sqlite::~Sqlite()
 
 gearmand_error_t Sqlite::initialize()
 {
-  gearmand::queue::Instance* exec_queue= new gearmand::queue::Instance(schema, table);
-
-  if (exec_queue == NULL)
-  {
-    return GEARMAND_MEMORY_ALLOCATION_FAILURE;
-  }
+  gearmand::queue::Instance* exec_queue = new gearmand::queue::Instance { schema, table };
```

Comment:
This one surprised me. I actually didn't know you could use a curly brace initializer list with `new`. Is that really the preferred syntax these days?

I get removing the antiquated check for NULL after calling the constructor in lines 93-96, but is anything set to catch the memory allocation exception if gearmand does run out of memory wherever this object is instantiated? I'm just wondering if we need a try/catch here instead.

Reply:
If there is any concern about curly braces I'll replace them. But actually it doesn't matter in this case.

```
#include <iostream>
#include <string>

class Foo {
    std::string _v{};
    int _a;
    public:
        Foo(std::string v, int a = 0) : _v(v), _a{a} {}
        std::string v(){ return _v; }
        int a() { return _a; }
        
};

int main()
{
    Foo x{"foo"};
    std::cout << "x v: " << x.v() << " a: " << x.a() << std::endl;

    Foo *y = new Foo("bar", 123);
    std::cout << "y v: " << y->v() << " a: " << y->a() << std::endl;

    Foo *z = new Foo{"baz", 321};
    std::cout << "z v: " << z->v()  << " a: " << z->a() << std::endl;

    return 0;
}
```

```
$g++ -o main *.cpp -Wall -std=c++0x
$main
x v: foo a: 0
y v: bar a: 123
z v: baz a: 321
```

Reply:
Hello, @p-alik. Please respond to this concern.

Reply:
I beg your pardon, @esabol for being irresponsible.
Any objection if we remove f39d20c0d1d8bc178db6731797539f666de8727e from the PR?
As to me we shouldn't spend more time on sqlite plugin at all. At least the feature of plugins is decided.

Reply:
> Any objection if we remove [f39d20c](https://github.com/gearman/gearmand/commit/f39d20c0d1d8bc178db6731797539f666de8727e) from the PR?

The fix seems easy to me and worth doing. Why not put `try { ... } catch (std::bad_alloc &) { return GEARMAND_MEMORY_ALLOCATION_FAILURE; }` around the `new`?

Reply:
It's done in f56b75c